scaaml/capture/scope/scope_template.py

```diff
@@ -28,7 +28,8 @@ def __init__(self):
         """Initialize the base."""
 
     @abstractmethod
-    def con(self,
+    def con(
```

Comment:
I think that for the abstraction, `def con(self, device_identifier: Optional[str] = None, **kwargs)` should be an appropriate signature.
`device_identifier` being a way to disambiguate which scope we want it case multiple scopes from the same model are plugged id and detected. Default behavior when set to `None` is to connect to the first one found.
You can decide to either put `**kwargs` as part of the signature or simply remove it because we don't need it at the moment for the scopes we support. When we'll need it it will be easy to add it anyway.

This means that we'll have to implement `def con()` for `ChipwhispererScope` so that it passes on the correct default values to `super().con()`.

Reply:
With your signature I am getting `error: Definition of "con" in base class "ScopeTemplate" is incompatible with definition in base class "OpenADC"`

Reply:
I think Python supports changing the signature when inheriting (`Pico6424E.con `signature is different from `ScopeTemplate.con` signature), but when doing multiple inheritance the signatures of superclasses must match. Other thing is that cw.capture_trace expects the argument scope to be of type Union[OpenADC, CWNano].

I do not understand the role of `ChipwhispererScope` in your last code snippet. We have a `CWScope` and `DefaultCWScope` context managers (auto-con/dis the `OpenADC`) and then we have a `Pico6424E` which plays a role of an object that behaves as an `OpenADC` (and is passed to cw API instead of `OpenADC`), then we have a context-manager with `PicoScope.scope` is `Pico6424E`. Finally we have reimplemented `ScopeTemplate` which is an abstract class capturing the cw scope interface.

Over all I agree with your comments in the sense of not polluting our API with unnecessary compatibility with CW. The thing is that I am not sure how to do that and still have a working typechecking. And I also agree that typechecking is far from perfect in the current PR (my idea was fix the workflows and then slowly improve the rest).